app/bulk_resource/views.py

```diff
@@ -41,438 +41,593 @@
+            if req_opt_desc.descriptor_id != -1:
+                descriptor = Descriptor.query.filter_by(
+                    id=req_opt_desc.descriptor_id
+                ).first()
+                if descriptor is not None and descriptor.name == form.required_option_descriptor.data:
+                    req_opt_desc_const = RequiredOptionDescriptorConstructor(
+                        name=descriptor.name,
+                        values=descriptor.values
+                    )
                     db.session.add(req_opt_desc_const)
-                    return redirect(url_for('bulk_resource.review_required_option_descriptor'))
-            # If descriptor is neither an existing descriptor nor a
-            # descriptor in the csv, output an error.
-            flash('Error: no such option descriptor. Please try again.', 'form-error')
-    # Collect all existing descriptors and descriptors in the csv to display
-    # as choices in the SelectField
+                    return redirect(url_for('bulk_resource.validate_required_option_descriptor'))
+            # If no descriptor found
+            flash('Error: No required option descriptor. Please try again.', 'form-error')
+
     descriptors = []
-    for header_cell in csv_container.csv_header_row.csv_header_cells:
-        if header_cell.descriptor_type == 'option':
-            descriptors.append(header_cell.data)
-    existing_descriptors = Descriptor.query.all()
-    for desc in existing_descriptors:
-        if desc.values:
-            descriptors.append(desc.name)
-    form.required_option_descriptor.choices = [(d, d) for d in descriptors]
     # If there is an existing required option descriptor, then make it
     # the default choice
-    req_opt_desc = RequiredOptionDescriptor.query.all()[0]
-    desc_name = ""
-    if req_opt_desc.descriptor_id != -1:
-        descriptor = Descriptor.query.filter_by(
-            id=req_opt_desc.descriptor_id
-        ).first()
-        if descriptor is not None:
-            desc_name = descriptor.name
-    form.required_option_descriptor.data = desc_name
+    req_name = ''
+    if csv_storage.action == 'update':
+        req_opt_desc = RequiredOptionDescriptor.query.all()[0]
+        if req_opt_desc.descriptor_id != -1:
+            descriptor = Descriptor.query.filter_by(
+                id=req_opt_desc.descriptor_id
+            ).first()
+            if descriptor is not None:
+                req_name = descriptor.name
+                descriptors.append(req_name)
+
+    # Collect all option descriptors in the CSV to display as choices in the
+    # SelectField.
+    for desc in csv_storage.csv_descriptors:
+        if desc.descriptor_type == 'option' and desc.name != req_name:
+            descriptors.append(desc.name)
+
+    # No existing required option descriptor and no option descriptors in CSV,
+    # then no possible required option descriptor
+    if len(descriptors) == 0:
```

Comment:
hm it's not enforced on app side but I'm not sure about if it should/should not be required? I feel like without one it is harder for the user to find resources? since without option descriptors, we would have no primary or secondary filters?